src/client/App.jsx

```diff
@@ -1,56 +1,110 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useEffect, useState } from "react";
+import "./App.css";
+import MovieForm from "./components/MovieForm";
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
-  const [movies, setMovies] = useState([]);
-
-  useEffect(() => {
-    fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
-  }, []);
-
-  const handleRegister = async ({ username, password }) => {
-    
-  };
-
-  const handleLogin = async ({ username, password }) => {
-    
-  };
-  
-  const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
-  }
-
-  return (
-    <div className="App">
-      <h1>Register</h1>
-      <UserForm handleSubmit={handleRegister} />
-
-      <h1>Login</h1>
-      <UserForm handleSubmit={handleLogin} />
-
-      <h1>Create a movie</h1>
-      <MovieForm handleSubmit={handleCreateMovie} />
-
-      <h1>Movie list</h1>
-      <ul>
-        {movies.map(movie => {
-          return (
-            <li key={movie.id}>
-              <h3>{movie.title}</h3>
-              <p>Description: {movie.description}</p>
-              <p>Runtime: {movie.runtimeMins}</p>
-            </li>
-          );
-        })}
-      </ul>
-    </div>
-  );
+	const [movies, setMovies] = useState([]);
+	const [registerResponse, setRegisterResponse] = useState(null);
+	const [loginResponse, setLoginResponse] = useState(null);
+	const [movieResponse, setMovieResponse] = useState(null);
+
+	useEffect(() => {
+		fetchMovies();
+	}, []);
+
+	const fetchMovies = async () => {
+		const res = await fetch(`${apiUrl}/movie`);
+		const data = await res.json();
+		setMovies(data.movies);
+	};
+
+	const handleRegister = async (user) => {
+		if (!user.username || !user.password) return;
+
+		const options = {
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify(user),
+		};
+
+		const res = await fetch(`${apiUrl}/user/register`, options);
+		const data = await res.json();
+		setRegisterResponse(data);
+	};
+
+	const handleLogin = async (user) => {
+		if (!user.username || !user.password) return;
+
+		const options = {
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify(user),
+		};
+
+		const res = await fetch(`${apiUrl}/user/login`, options);
+		const accessToken = await res.json();
+
+		localStorage.setItem("access-token", accessToken.data);
+		setLoginResponse(accessToken);
+	};
+
+	const handleCreateMovie = async (movie) => {
```

Comment:
the logic of your handleCreatedMovie flows really nicely. i dont get all the syntax but i can assume/guess what each sections does because of the clear flow

Reply:
Part of why I like using async/await is it's readability, glad you agree!